tests/logger_logging/test_log_forwarding.py

```diff
@@ -0,0 +1,93 @@
+# Copyright 2010 New Relic, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import logging
+
+from newrelic.api.background_task import background_task
+from newrelic.api.time_trace import current_trace
+from newrelic.api.transaction import current_transaction
+from testing_support.fixtures import reset_core_stats_engine
+from testing_support.validators.validate_log_event_count import validate_log_event_count
+from testing_support.validators.validate_log_event_count_outside_transaction import validate_log_event_count_outside_transaction
+from testing_support.validators.validate_log_events import validate_log_events
+from testing_support.validators.validate_log_events_outside_transaction import validate_log_events_outside_transaction
+
+
+def set_trace_ids():
+    txn = current_transaction()
+    if txn:
+        txn._trace_id = "abcdefgh12345678"
+    trace = current_trace()
+    if trace:
+        trace.guid = "abcdefgh"
+
+def exercise_logging(logger):
+    set_trace_ids()
+
+    logger.debug("A")
+    logger.info("B")
+    logger.warning("C")
+    logger.error("D")
+    logger.critical("E")
+    
+    assert len(logger.caplog.records) == 3
+
+
+_common_attributes_service_linking = {"timestamp": None, "hostname": None, "entity.name": "Python Agent Test (logger_logging)", "entity.guid": None}
+_common_attributes_trace_linking = {"span.id": "abcdefgh", "trace.id": "abcdefgh12345678", **_common_attributes_service_linking}
+
+_test_logging_inside_transaction_events = [
+    {"message": "C", "level": "WARNING", **_common_attributes_trace_linking},
+    {"message": "D", "level": "ERROR", **_common_attributes_trace_linking},
+    {"message": "E", "level": "CRITICAL", **_common_attributes_trace_linking},   
```

Comment:
These aren't going to fly with Py2, need to call `.update(_common_attributes_whatever)` on all of these.